lexibank_chaconarawakan.py

```diff
@@ -39,41 +39,106 @@ def cmd_install(self, **kw):
             "Aikhenvald 2001": "aikhenvald_dicionario_2001",
         }
 
-        wl = lingpy.Wordlist(self.raw.posix("arawakan_swadesh_100_edictor.tsv"))
+        # add source
+        args.writer.add_sources()
 
-        with self.cldf as ds:
-            concepts = {}
-            ds.add_sources(*self.raw.read_bib())
+        # add languages
+        languages = args.writer.add_languages(lookup_factory="Name")
 
-            for l in self.languages:
-                ds.add_language(ID=slug(l["Name"]), Name=l["Name"], Glottocode=l["Glottocode"])
-            for concept in self.conceptlist.concepts.values():
-                ds.add_concept(
-                    ID=concept.id,
-                    Name=concept.english,
-                    Concepticon_ID=concept.concepticon_id,
-                    Concepticon_Gloss=concept.concepticon_gloss,
-                    Portuguese_Gloss=concept.attributes["portuguese"],
-                )
-                concepts[slug(concept.english)] = concept.id
+        # add concepts
+        concepts = args.writer.add_concepts(
+            id_factory=lambda c: "%s_%s"
+            % (c.id.split("-")[-1], slug(c.english)),
+            lookup_factory="Name",
+        )
 
-            for k in pb(wl, desc="wl-to-cldf"):
-                if wl[k, "value"]:
-                    for row in ds.add_lexemes(
-                        Language_ID=slug(wl[k, "doculect"]),
-                        Parameter_ID=concepts[slug(wl[k, "concept"])],
-                        Value=wl[k, "value"],
-                        Form=wl[k, "form"],
-                        Segments=wl[k, "segments"],
-                        Source=src.get(wl[k, "source"], ""),
-                    ):
+        # Hard-coded fixes to segment errors in raw source
+        segments = {
```

Comment:
Are you going to replace this with the new `segments.csv` based mechanism?

Reply:
yes, it would be a good test, @tresoldi

Reply:
I've just pushed it with f04b60e; `FORMS.md` tells me that there were 63 replacements. If they are too many, we need a profile...

BTW, `--dev` is a much welcomed addition.

Reply:
I had forgotten to add `segments.csv`, it is there with cb4c089